extra/regex/Makefile

```diff
@@ -0,0 +1,3 @@
+build:
```

Comment:
The `build` target should be done in a similar way to the `compiler` directory, so it does nothing if nothing needs to be built.

There should be a `push` target in the compiler directory, with a empty file in the build directory whose timestamp says tells whether the push is up to date. The `build` target should depend on this timestamp file.

You need to control the build directory as the compiler Makefile does.

Reply:
Fixed such that all build artifacts (`bala` file and regex `jar` file) will end up in separate subdirectories of the `./build` directory. I am using the `bala` file itself as the timestamp file. I assumed that we will have multiple extras in the future and will prefer to be able to build them separately (with the option to build all of them together)